cmd/codesign/main.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+var entitlements = `<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>com.apple.security.virtualization</key>
+	<true/>
+</dict>
+</plist>`
+
+func main() {
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run() error {
+	f, err := os.CreateTemp("", "*.entitlements")
+	if err != nil {
+		return fmt.Errorf("failed to create temp file: %w", err)
+	}
+	defer f.Close()
+
+	if _, err := f.WriteString(entitlements); err != nil {
+		return fmt.Errorf("failed to write entitlements content: %w", err)
+	}
+
+	if err := f.Close(); err != nil {
+		return fmt.Errorf("failed to close temp file: %w", err)
+	}
+	defer os.Remove(f.Name()) // clean up
+
+	cmd := exec.Command("codesign", "--entitlements", f.Name(), "-s", "-", os.Args[1])
```

Comment:
use a variable `codesignArgs := []string { "codesign", "--entitlements", f.Name(), "-s", "-", os.Args[1]} ` to avoid code duplication when there is an error?

Reply:
I think it is not a large benefit. It's not a part of the system that is frequently modified.